File: Data Structures & Algorithms/target-sum/submission-0.py
class Solution:
    def findTargetSumWays(self, nums: List[int], target: int) -> int:
        n = len(nums)
        output = [0]
        for i in nums:
            temp = []
            for j in output:
                temp.append(j+i)
                temp.append(j-i)
            output = temp
        
        # print(output)
        output = Counter(output)
        return output[target]

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        // increasing order
        unordered_map<int,int> temp;
        for(int i = 0;i<numbers.size();i++){
            // std::cout << i << ' ' << numbers[i] << ' ';
            if(temp.contains(target-numbers[i])){
                return {temp[target-numbers[i]]+1,i+1};
            }

            temp[numbers[i]] = i;
        }
        return {-1,-1};
    }
};
